tests/integration/test_jdbc.py

```diff
@@ -0,0 +1,155 @@
+        {
+            "name": "TABLE_SCHEM",
+            "type": "varchar",
+            "typeSignature": {"rawType": "varchar", "arguments": []},
+        }
+    ]
+    assert len(result["data"]) == 3
+    assert result["data"] == [
+        ["", "root"],
+        ["", "a_schema"],
+        ["", "system_jdbc"],
+    ]
+
+
+def test_jdbc_has_table(app_client, c):
+    create_meta_data(c)
+    check_data(app_client)
+
+    response = app_client.post("/v1/statement", data="SELECT * from system.jdbc.tables")
+    assert response.status_code == 200
+    result = get_result_or_error(app_client, response)
+
+    assert "columns" in result
+    assert "data" in result
+    assert len(result["data"]) == 4
+    assert result["data"] == [
+        ['', 'a_schema', 'a_table', '', '', '', '', '', '', ''],
+        ['', 'system_jdbc', 'schemas', '', '', '', '', '', '', ''],
+        ['', 'system_jdbc', 'tables', '', '', '', '', '', '', ''],
+        ['', 'system_jdbc', 'columns', '', '', '', '', '', '', '']
+    ]
+
+
+def test_jdbc_has_columns(app_client, c):
+    create_meta_data(c)
+    check_data(app_client)
+
+    response = app_client.post("/v1/statement", data=f"SELECT * from system.jdbc.columns where TABLE_NAME = '{table}'")
+    assert response.status_code == 200
+    result = get_result_or_error(app_client, response)
+
+    assert "columns" in result
+    assert "data" in result
+    assert len(result["data"]) == 3
+    assert result["data"] == [
+        ['', 'a_schema', 'a_table', 'A_STR', 'VARCHAR', 'VARCHAR', '', '', '', '', '', '', '', 'VARCHAR', '', '', '1', '', '', '', '', '', '', ''],
+        ['', 'a_schema', 'a_table', 'AN_INT', 'INTEGER', 'INTEGER', '', '', '', '', '', '', '', 'INTEGER', '', '', '2', '', '', '', '', '', '', ''],
+        ['', 'a_schema', 'a_table', 'A_FLOAT', 'FLOAT', 'FLOAT', '', '', '', '', '', '', '', 'FLOAT', '', '', '3', '', '', '', '', '', '', ''],
+    ]
+
+
+def create_table_row(a_str: str = "any", an_int: int = 1, a_float: float = 1.1):
+    return {
+        "A_STR": a_str,
+        "AN_INT": an_int,
+        "A_FLOAT": a_float,
+    }
+
+
+def check_data(app_client):
+    response = app_client.post("/v1/statement", data=f"SELECT * from {schema}.{table}")
+    assert response.status_code == 200
+    a_table = get_result_or_error(app_client, response)
+    assert "columns" in a_table
+    assert "data" in a_table
+    assert "error" not in a_table
+
+
+def get_result_or_error(app_client, response):
+    result = response.json()
+
+    assert "nextUri" in result
+    assert "error" not in result
+
+    status_url = result["nextUri"]
+    next_url = status_url
+
+    counter = 0
+    while True:
+        response = app_client.get(next_url)
+        assert response.status_code == 200
+
+        result = response.json()
+
+        if "nextUri" not in result:
+            break
+
+        next_url = result["nextUri"]
+
+        counter += 1
+        assert counter <= 100
+
+        sleep(0.1)
+
+    return result
```

Comment:
I come from a Java world where the convention is to put "private" code at the end but actually coped from test_server.py which has it at the end. Shall I re-use from test_server.py? @charlesbluca

Reply:
Ah I didn't realize that this was also the case in test_server.py; in that case my preference would be to move the util functions up to the top in both files - my reasoning here being to match up with other testing files (such as [test_compatibility.py](https://github.com/dask-contrib/dask-sql/blob/main/tests/integration/test_compatibility.py)) that place their utility functions at the top of the file.

I also note that since we are duplicating functions like `get_result_or_error`, we might want to add a testing utility module as a central location for functions like this, but that is out of the scope of this PR.